### DISCUSSION TITLE
Exposing unused fields in FastAPI docs

### DISCUSSION THREAD

**Discussion by robz-tirtlib** [fastapi-users/fastapi-users#1371] [Q&A]
Inheritance from this (https://github.com/fastapi-users/fastapi-users/blob/ae9f52474ba2c7baebeb923e4d03aea479765362/fastapi_users/schemas.py#L64) Base class causes fields is_active, is_superuser, is_verified to be exposed in /register endpoint auto-generated FastAPI doc. But they are not being involved in the process of registration. So it causes confusion for front-end implementing client side.

My guess on why it is made this way - to support manual creation of users by admins of system. If it is the case why not implement two separate flows for these actions (and separate schemas as well)?

Otherwise, what was the point?

Thanks in advance, just trying to understand the motivation behind this solution.

**Comment by ggurdanikita** [comment]
I also encountered this, I had to test and make sure that these fields are really not used.

Still, you made an amazing library!

**Comment by frankie567** [comment]
I confirm that those fields are ignored in `/register` and other self-service endpoints.

The main motivation behind this is for simplicity. Otherwise, we would need to have a dedicated schema for `/register`. But maybe that could be revisited or generated automatically.
